### PR TITLE
[stable/traefik] Allow static IP address

### PR DESCRIPTION
**Pull request by dhcdata**
If cloud provider supports, will assign the static IP address to the raefik service https://kubernetes.io/docs/api-reference/v1.6/#servicespec-v1-core

Confirmed on GCP.

### CODE REVIEW

**Comment by dhcdata** [comment]
CLA sent

**Comment by dhcdata** [comment]
CLA sent

**Comment by dhcdata** [comment]
New Comment

File: stable/traefik/templates/service.yaml [hunk 1]
```diff
@@ -30,3 +30,6 @@ spec:
     {{- if not .Values.ssl.enabled }}
     targetPort: 80
     {{- end }}
+  {{- if .Values.loadBalancerIP}}
```
Comment (unguiculus) [inline stable/traefik/templates/service.yaml]: Nit: Add space before `}}`.

**Review by unguiculus — CHANGES_REQUESTED** [review]
One small thing. Otherwise LGTM.

**Comment by unguiculus** [comment]
Check you CLA again. Did you commit with the correct e-mail?

**Comment by dhcdata** [comment]
Done

File: stable/traefik/values.yaml [hunk 2]
```diff
@@ -2,6 +2,8 @@
 image: traefik
 imageTag: v1.2.1
 serviceType: LoadBalancer
+# Static IP Address pass --set loadBalancerIP=false from command line if running mulitiple
```
Comment (unguiculus) [inline stable/traefik/values.yaml]: `loadBalancerIP=false` doesn't seem to make sense. Either set it or leave it empty. Also, there's a typo in `mulitiple`. I'd just remove the comment.

**Review by unguiculus — CHANGES_REQUESTED** [review]

**Comment by dhcdata** [comment]
Because helm supports multiple release objects, if you set `loadBalancerIP` in the `values.yaml` file and run `helm install .` followed by `helm install --name my-project`, the second release object will never obtain an external IP address since it will keep attempting to assign itself to one that is taken.  If needs to be overridden with `false` from the command line to obtain a dynamic IP address or set to another IP address

Perhaps the better approach is to:
1) remove the comment from `values.yaml`
2) add documentation to README.md
3) uncomment `loadBalancerIP` and set the default value to false

@unguiculus thoughts?

**Comment by unguiculus** [comment]
A value of `false` doesn't make sense. Just set the default value to empty in `values.yaml`. If you install the chart twice, it is, of course, your responsibility to configure each installation properly.

**Comment by dhcdata** [comment]
The problem is that the vast majority that would utilize the `loadBalancerIP`  would set their static IP address in `values.yaml`. For the small minority that would have multiple release objects, the command line flag `--set` doesn't allow `empty` to override `values.yaml`.  

I've updated to have the default be `loadBalacerIP: dynamic`.  The template will now check that the values do not equal `nil` or `dynamic` before attempting the static IP.

**Comment by unguiculus** [comment]
The problem you are making up just does not exist. Please set the default to empty in `values.yaml`. If someone installs the chart multiple times, they will have to provide different values for each installation, either using `--set` or different `values.yaml` files. Look e. g. at this example to see how it should be done:

https://github.com/kubernetes/charts/blob/master/stable/nginx-ingress/templates/controller-service.yaml#L21-L23
https://github.com/kubernetes/charts/blob/master/stable/nginx-ingress/values.yaml#L101

Example for installing the chart twice with different IPs:
```
helm install stable/traefik --name rel1 --set loadBalancerIP=10.20.30.40
helm install stable/traefik --name rel2 --set loadBalancerIP=10.20.30.41
```

**Comment by dhcdata** [comment]
done

**Comment by unguiculus** [comment]
Please update the service as well.

**Pull request closed without merging**